### ISSUE TITLE
Clang cpp always passes `-traditional-cpp`

### ISSUE THREAD

**Issue by gjurgensen** [llvm/llvm-project#94675]
When `cpp` wraps `clang`, it always passes the flag `-traditional-cpp`, which prevents full preprocessing of modern C code. As far as I can tell, there is no way to disable or overrule the `-traditional-cpp` flag.

This manifest in the following example. Provided the following `main.c` file:
```c
#ifndef MY_MACRO
  #define MY_MACRO 1
#endif

int main(void) {
  return 0;
}
```
Preprocessing with the command `cpp -P main.c` yields the following:
```c

  #define MY_MACRO 1


int main(void) {
  return 0;
}

```
The issue here is that the output of `cpp` is not fully preprocessed as expected.

Running `cpp -E -P main.c` works as expected, as it does not pass the `-traditional-cpp` flag. It seems to me there should be a way to disable the usage of this flag from `cpp` (for reference, the `gcc`-version of `cpp` does not pass the `-traditional-cpp` flag by default).

This was observed on the following version:
```shell
Apple clang version 15.0.0 (clang-1500.3.9.4)
Target: arm64-apple-darwin23.5.0
```

**Comment by efriedma-quic** [comment]
The "cpp" wrapper isn't part of LLVM; it's Apple proprietary code.  Please report directly to Apple. (I think currently the preferred way is the Feedback Assistant?)
